iron-overlay-behavior.d.ts

```diff
@@ -0,0 +1,263 @@
+     */
+    _onCaptureEsc(event: Event): any;
+
+    /**
+     * Handles TAB key events to track focus changes.
+     * Will wrap focus for overlays withBackdrop.
+     */
+    _onCaptureTab(event: Event): any;
+
+    /**
+     * Refits if the overlay is opened and not animating.
+     */
+    _onIronResize(): any;
+
+    /**
+     * Will call notifyResize if overlay is opened.
+     * Can be overridden in order to avoid multiple observers on the same node.
+     */
+    _onNodesChange(): any;
+  }
+
+  const IronOverlayBehaviorImpl: object;
+
+  /**
+   *   Use `Polymer.IronOverlayBehavior` to implement an element that can be hidden or shown, and displays
+   *   on top of other content. It includes an optional backdrop, and can be used to implement a variety
+   *   of UI controls including dialogs and drop downs. Multiple overlays may be displayed at once.
+   *
+   *   See the [demo source code](https://github.com/PolymerElements/iron-overlay-behavior/blob/master/demo/simple-overlay.html)
+   *   for an example.
+   *
+   *   ### Closing and canceling
+   *
+   *   An overlay may be hidden by closing or canceling. The difference between close and cancel is user
+   *   intent. Closing generally implies that the user acknowledged the content on the overlay. By default,
+   *   it will cancel whenever the user taps outside it or presses the escape key. This behavior is
+   *   configurable with the `no-cancel-on-esc-key` and the `no-cancel-on-outside-click` properties.
+   *   `close()` should be called explicitly by the implementer when the user interacts with a control
+   *   in the overlay element. When the dialog is canceled, the overlay fires an 'iron-overlay-canceled'
+   *   event. Call `preventDefault` on this event to prevent the overlay from closing.
+   *
+   *   ### Positioning
+   *
+   *   By default the element is sized and positioned to fit and centered inside the window. You can
+   *   position and size it manually using CSS. See `Polymer.IronFitBehavior`.
+   *
+   *   ### Backdrop
+   *
+   *   Set the `with-backdrop` attribute to display a backdrop behind the overlay. The backdrop is
+   *   appended to `<body>` and is of type `<iron-overlay-backdrop>`. See its doc page for styling
+   *   options.
+   *
+   *   In addition, `with-backdrop` will wrap the focus within the content in the light DOM.
+   *   Override the [`_focusableNodes` getter](#Polymer.IronOverlayBehavior:property-_focusableNodes)
+   *   to achieve a different behavior.
+   *
+   *   ### Limitations
+   *
+   *   The element is styled to appear on top of other content by setting its `z-index` property. You
+   *   must ensure no element has a stacking context with a higher `z-index` than its parent stacking
+   *   context. You should place this element as a child of `<body>` whenever possible.
+   *
+   *   
+   */
+  interface IronOverlayBehavior {
+  }
+
+  const IronOverlayBehavior: object;
```

Comment:
how is it expressed that `IronOverlayBehavior` is composed by other behaviors?
https://github.com/PolymerElements/iron-overlay-behavior/blob/master/iron-overlay-behavior.html#L796

Reply:
Huh, that's a good catch. We don't handle behavior arrays right now. Filed https://github.com/Polymer/gen-typescript-declarations/issues/80 and will update this PR when done.

Reply:
Updated to add the other behaviors. See interface above.